src/index.ts

```diff
@@ -0,0 +1,189 @@
+import type appdmgType from 'appdmg';
+import debugCreator from 'debug';
+import { existsSync, promises as fs } from 'fs';
+import * as path from 'path';
+
+const debug = debugCreator('electron-installer-dmg');
+
+export type ElectronInstallerDMGOptions = {
+  /**
+   * The `.app` directory generated by Electron Packager.
+   */
+  appPath: string;
+  /**
+   * The application name.
+   */
+  name: string;
+  /**
+   * The title of the produced DMG, which will be shown when mounted.
+   *
+   * Defaults to the application name.
+   */
+  title?: string;
+  /**
+   * Path to the background for the DMG window. Background image should be of size 658x498.
+   */
+  background?: string;
+  /**
+   * Path to the icon to use for the app in the DMG window.
+   */
+  icon?: string;
+  /**
+   * How big to make the icon for the app in the DMG. [Default: `80`].
+   */
+   iconSize?: number;
```

Comment:
Is there linting set up for this repo? Seems like there's an extra space on this line.
```suggestion
  iconSize?: number;
```